reactive/filebeat.py

```diff
@@ -112,8 +112,13 @@ def push_filebeat_index(elasticsearch):
     hosts = elasticsearch.list_unit_data()
     for host in hosts:
         host_string = "{}:{}".format(host['host'], host['port'])
-    push_beat_index(host_string, 'filebeat')
-    set_state('filebeat.index.pushed')
+    if push_beat_index(elasticsearch=host_string,
+                       service='filebeat', fatal=False):
+        set_state('filebeat.index.pushed')
+        status.active('Filebeat ready.')
+    else:
+        # curl'ing ES may fail if ES isnt quite ready; keep trying
+        status.waiting('Attempting to push filebeat index to Elasticsearch.')
```

Comment:
I generally don't like relying on the `update-status` hook to do automatic retries, as that's not the intent of the hook and the charm has no control over when it happens; not to mention the 5 minute period seems overly long for most cases.  There was even talk about making `update-status` one of the "restricted hooks" like `collect-metrics` and `meter-status-changed`, so that only the `@hook('update-status')` handler would fire and nothing else, though I think at this point that would break too many charms.

At any rate, I'd much rather see an explicit retry loop here with a cutoff after a certain number of attempts and possibly a scaling delay.  That, or some flag sent over the relation notifying services when they can push, rather than forcing them to poll like this.

Reply:
@johnsca i've added a commit that limits the number of retries.  If exhausted, we'll put the charm in blocked state, but I don't think we should fail the filebeat hook.  Blocking feels right to me to let the operator know something is goofy with ES.